test/Microsoft.Azure.Devices.Edge.Test/helpers/Context.cs

```diff
@@ -74,7 +75,7 @@ public Context()
                 return Option.None<(string, string, string)>();
             }
 
-            string GenerateDefaultDeviceId() =>
+            string defaultId =
                 $"e2e-{string.Concat(Dns.GetHostName().Take(15))}-{DateTime.Now:yyMMdd'-'HHmmss'.'fff}";
 
             const int DeviceIdMaxLength = 37;
```

Comment:
In some tests we create leaf devices whose identities are derived from the edge device's identity. Then we create certificates with Common Names (CNs) derived from the leaf device's identity. So we have:

edge device identity: `myedgeid`
leaf device identity: `myedgeid-myleafid`
CN: `myedgeid-myleafid-mycnid`

CNs must be < 64 chars, so we need to be careful about the uplevel names we create to ensure there's enough room to create downlevel names.

There's a comment about this in the `TransparentGateway` test.

Reply:
I see. Could you please add a comment for the variable citing its constraint with the `TransparentGateway` Test please?

Reply:
I just pushed a new class `IdentityLimits` that encapsulates all these identity length limits, with an appropriate comment and static check methods. Unfortunately the building of these identity strings is not consolidated into this new class because that would be rather difficult. But you can easily search the codebase for `IdentityLimits` to see all the method callers; in each case the string building logic is right there next to the call.